api/desecapi/views.py

```diff
@@ -120,6 +125,90 @@ class DomainDetailByName(DomainDetail):
     lookup_field = 'name'
 
 
+class RRsetDetail(generics.RetrieveUpdateDestroyAPIView):
+    lookup_field = 'type'
+    serializer_class = RRsetSerializer
+    permission_classes = (permissions.IsAuthenticated, IsDomainOwner,)
+    restricted_types = ('SOA', 'RRSIG', 'DNSKEY', 'NSEC3PARAM')
+
+    def delete(self, request, *args, **kwargs):
+        try:
+            super().delete(request, *args, **kwargs)
+        except Http404:
+            pass
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def get_queryset(self):
+        name = self.kwargs['name']
+        subname = self.kwargs['subname'].replace('=2F', '/')
+        type_ = self.kwargs['type']
+
+        # This may be more cleanly implemented using object level permissions.
```

Comment:
This is harder than it seems, see https://github.com/desec-io/desec-stack/pull/50#discussion_r124585051
Bottom line: Don't use permissions here. I removed the comment.